### PR TITLE
mention code formatting in contribution guidelines

### PR DESCRIPTION
**Pull request by expipiplus1**
Closes https://github.com/shader-slang/slang/issues/5511

### CODE REVIEW

File: docs/design/coding-conventions.md [hunk 1]
```diff
@@ -136,165 +136,20 @@ All standard module code that a Slang user might link against should go in the `
...
-For consistentcy, a trailing `\` can be used, followed by a comment to mark the end of the definition:
-
-```c++
-#define FOREACH_ANIMAL(X)   \
-    X(Aardvark)             \
-    X(Bat)                  \
-    /* end */
-```
+If you open a pull request and the formatting is incorrect, you can comment
+`/format` and a bot will format your code for you.
```
Comment (csyonghe) [inline docs/design/coding-conventions.md:152]: We should explain that a PR will be created and you need to merge that PR.

**Comment by jkwak-work** [comment]
Is there a way to run the formatting my local machine? I personally prefer to get is done before presenting it to others as a review.

File: CONTRIBUTION.md [hunk 2]
```diff
@@ -202,7 +202,10 @@ $ git push origin feature/your-feature-name
 ```
 
 ### Request Pull
-Request a pull from "shader-slang" repository.
+
+Open a pull request against `shader-slang/slang`.
+
+Code formatting can be automatically fixed on your branch by commenting `/format`, a bot will proceed to open a PR targeting *your* branch.
```
Comment (jkwak-work) [inline CONTRIBUTION.md:208]: The formatting is a required process not an optional step. I think we should describe that the generated PR needs to be merged here as Yong pointed out on the other spot.

**Review by jkwak-work — APPROVED** [review]
Looks good to me.

**Comment by csyonghe** [comment]
@jkwak-work You can configure VisualStudio or VSCode to perform formatting on save.

In visual studio it is called Code Cleanup:
image

You also need to set code formatter to clang-format.

**Review by csyonghe — APPROVED** [review]

**Pull request merged**